stock_manager/cli.py

```diff
@@ -0,0 +1,558 @@
+"""
+Command Line Interface builder and handler for the Stock Manager Application
+
+Provides a CLI for performing various operations across the application without needing a GUI.
+"""
+
+import argparse
+import os
+import sys
+from typing import Callable
+
+import pytest
+
+import stock_manager
+
+
+def build_commands() -> argparse.Namespace:
+    """
+    Builds and parses command-line arguments for the CLI interface.
+    
+    :return: Parsed command-line arguments including flags and values.
+    """
+    
+    parser = argparse.ArgumentParser(description='Common Stock Manager CLI')
+    
+    # core arguments
+    flags = [
+        ('-r', '--run', 'Runs Application GUI, Same As Running "python -m stock_manager"'),
+        ('-v', '--version', 'Prints stock_manager\'s git version'),
+        ('-t', '--test', r'Runs All Tests Located In stock_manager\tests Using PyTest'),
+        ('-e', '--export', 'Exports All Item Data To A Specified File Type And Location '
+                           '(Used With Path And Extension Arguments)'),
+        ('-qr', '--create-qr', 'Generates A QR Code Of A Specified Item And Stores It As A '
+                               '.PNG In A Specified Location (Used With Path And Part Number Arguments)'),
+        ('-s', '--sync-databases', 'Synchronize Both Databases, Setting The Values Of The SQL Database '
+                                   'To That Of The Google Sheet (Only Run If A Database Is Changed Externally)'),
+        ('-li', '--list-items', 'Lists All Items In The Database'),
+        ('-si', '--search-items', 'Searches For A Specified Value In All Fields In All Items'),
+        ('-ai', '--add-item', 'Add An Item With Specified Values To Both Databases '
+                              '(Used With Sequence Of 10 Item Detail Arguments)'),
+        ('-ri', '--remove-item', 'Remove A Specified Item From Both Databases After Confirmation Query'),
+        ('-lu', '--list-users', 'Lists All Users In The Database'),
+        ('-su', '--search-users', 'Searches For A Specified Username In All Usernames'),
+        ('-au', '--add-user', 'Add A User With Specified Username To Both Databases'),
+        ('-ru', '--remove-user', 'Remove A Specified User From Both Databases After Confirmation Query'),
+        ('-ei', '--edit', 'Edit A Specified Item With At Least One Field Name And New Value')
+    ]
+    
+    for short, long, desc in flags:
+        parser.add_argument(short, long, action='store_true', help=desc)
+    
+    # optional positional arguments
+    parser.add_argument(
+            'values',
+            nargs='*',
+            default=None,
+            help='A Sequence Of Optional Positional Arguments'
+    )
```

Comment:
I think you're robbing yourself of the opportunity to document which additional args are required for each option by laying them out as if they are all always the same. You're also allowing for the user to make mistakes and not realize what they've done.